Libraries/Opc.Ua.Gds.Server.Common/CertificateGroup.cs

```diff
@@ -72,17 +74,13 @@ public virtual async Task Init()
                 X509Certificate2Collection certificates = await store.Enumerate();
                 foreach (var certificate in certificates)
                 {
-                    if (Utils.CompareDistinguishedName(certificate.Subject, m_subjectName))
+                    if (X509Utils.CompareDistinguishedName(certificate.Subject, m_subjectName))
                     {
-                        using (RSA rsa = certificate.GetRSAPublicKey())
+                        if (X509Utils.GetRSAPublicKeySize(certificate) != Configuration.CACertificateKeySize)
                         {
-                            if (rsa.KeySize != Configuration.CACertificateKeySize)
-                            {
-                                continue;
-                            }
-
-                            // TODO check hash size
+                            continue;
                         }
+                        // TODO check hash size
```

Comment:
I thing it would be good, to address the TODO with this change

Reply:
better in the GDS work item, here it is unrelated and may require a test case